doc/handle-ASIC-SDK-health-event/handle-ASIC-SDK-health-event.md

```diff
@@ -0,0 +1,635 @@
+ *    "severity": "2",
+ *    "timestamp": {
+ *        "tv_sec": "22429",
+ *        "tv_nsec": "3428724"
+ *    },
+ *    "category": "3",
+ *    "data": {
+ *        data_type: "0"
+ *    },
+ *    "additional_data": "Some additional information"
+ * }
+ */
+
+typedef void (*sai_switch_asic_sdk_health_event_notification_fn)(
+        _In_ sai_object_id_t switch_id,
+        _In_ sai_switch_asic_sdk_health_severity_t severity,
+        _In_ sai_timespec_t timestamp,
+        _In_ sai_switch_asic_sdk_health_category_t category,
+        _In_ sai_switch_health_data_t data,
+        _In_ const sai_u8_list_t description);
+
+```
+
+The following type definitions for the SAI attributes defined in `SAI/inc/saitypes.h` are used in this document.
+
+```C
+typedef enum _sai_health_data_type_t
+{
+    /** General health data type */
+    SAI_HEALTH_DATA_TYPE_GENERAL
+} sai_health_data_type_t;
+
+typedef struct _sai_switch_health_data_t
+{
+    /** Type of switch health data */
+    sai_health_data_type_t data_type;
+
+} sai_switch_health_data_t;
+```
+
+### Configuration and management
+
+#### Manifest (if the feature is an Application Extension)
+
+N/A.
+
+#### CLI Enhancements
+
+##### Configure suppress ASIC/SDK health events by severity and category
+
+Command `config asic-sdk-health-event suppress <severity> <category-list>|<none>|<all>` is introduced for a customer to configure the categories that he/she wants to suppress for a certain severity.
+
+The severity can be one of `fatal`, `warning`, and `notice`.
+
+The category-list is a list whose element is one of `software`, `firmware`, `cpu_hw`, `asic_hw` separated by a comma. The order does not matter.
+
+If the category-list is `none`, none category is suppressed and all the categories will be notified for `severity`.
+
+If the category-list is `all`, all the categories are suppressed and none category will be notified for `severity`.
+
+The ASIC/SDK health events whose `category` is in `category-list` with the `severity` will not be reported by the vendor SAI once the corresponding SAI attributes are set.
+But the events that were reported after the command is executed but before the SAI attributes are set will be handled by orchagent and pushed into `STATE_DB.ASIC_SDK_HEALTH_EVENT_TABLE` as usual.
+
+Eg. the following command will suppress `notice` events with category `asic_hw` and `cpu_hw`:
+
+`config asic-sdk-health-event suppress notice asic_hw,cpu_hw`
+
+The following error message will be shown if a customer configures it on a platform that does not support it.
+
+`ASIC/SDK health event is not supported on the platform`
+
+The following error message will be shown if a customer suppresses a severity which is not supported on the platform.
+
+`Suppressing ASIC/SDK health {severity} event is not supported on the platform`
+
+##### Display the ASIC/SDK health events
+
+Command `show asic-sdk-health-events received` is introduced to display the ASIC/SDK health events as a table.
```

Comment:
`show asic-sdk-health-events received`  - `received `is not an option right?

Reply:
Correct. It's a subcommand. We have two subcommands, `show asic-sdk-health-event received` and `show asic-sdk-health-event suppress-category-list`